### DISCUSSION TITLE
.env file not loading if added in .gitignore file

### DISCUSSION THREAD

**Discussion by DeeshanSharma** [goatandsheep/react-native-dotenv#337] [Q&A]
I discovered a strange bug that if I git ignore any env file and create a production apk the env variables give undefined while they work perfectly in the development. I have done all the configuration correctly as mentioned in the docs. Found this issue after spending two working days in my office. Help me through this or guide me if I am wrong anywhere as env files could not be commited on github.

**Comment by goatandsheep** [comment]
depends on how you're building the production APK. The platform you're using should have the option to add environment variables.

**Reply by DeeshanSharma** [reply]
I am using eas build locally on wsl ubuntu jammy and expo managed workflow. If you need any file or logs let me know

**Reply by goatandsheep** [reply]
Make sure to keep your `.env` files in your `.gitignore`. It is a major security risk to commit. [The EAS documentation says](https://docs.expo.dev/build-reference/variables/#secrets-on-the-expo-website):

>To create app-specific secrets, navigate to the "Secrets" tab in your project dashboard.
